.buildkite/Dockerfile.gpu

```diff
@@ -61,6 +63,9 @@ COPY . .
 RUN ./ci/travis/ci.sh init
 RUN bash --login -i ./ci/travis/ci.sh build
 
+RUN (if [ "${INSTALL_DEPENDENCIES}" = "ML" ]; then RLLIB_TESTING=1 TRAIN_TESTING=1 TUNE_TESTING=1 bash --login -i ./ci/travis/install-dependencies.sh; fi)
+RUN (if [ "${INSTALL_DEPENDENCIES}" = "ML" ]; then pip install -Ur ./python/requirements_ml_docker.txt; fi)
```

Comment:
I'm of the opinion that we should try to keep dependency installation to be the minimal set needed for each test suite.

If we always install the superset of dependencies, our tests may not be reflective of real user setups and may miss some cases. Though perhaps for GPU we do intentionally want to include at least the ML docker image dependencies.... thoughts?

Reply:
The goal of moving the installs into the Dockerfile is to speedup subsequent pipeline steps, e.g. reduce the cumulative installation time. The dependencies that benefit the most from this are presumably large dependencies, like tensorflow and torch.

The decision we'll have to make here is if we care about testing a minimal setup, and if we want to include these checks in the CI pipeline. We do have a minimal dependency test step, but it only tests a small subset at the moment.

That said, I agree that we should probably limit the default dependencies here.

Actually, let's hold off this PR for now, I think I have an idea on how to generally solve this problem.